Flaconi/terraform-aws-cdn
# OPS-4735 added optional Cloud Front functions

# Optional Cloud Front functions

## Description
This PR adds optional Cloud Front functions creation and association with the created Cloud Front distribution so that we would be able to associate some logic to the CF request and response.

## Testing Instructions
The module is used to create a cdn to serve the product feeds generated by Productsup and our custom lambda function in the following lite-infra branch. The optional function creation would enable us to add CF functions to the distribution. Please see the usage here and it has been deployed to dev.

https://github.com/Flaconi/lite-infra/blob/PIMF-1502/terragrunt/envs/aws/dev/eu-central-1/applications/ct-product-feed/feed-cdn/terragrunt.hcl

## How to roll out
After PR approval the release tag is created and the relevant tag is updated in the branch PIMF-1502 of lite-infra repository.

## Demo
The CF function enables basic http auth for the following cdn url:
(https://product-feed.dev.flaconi-platform.net/export/pl_raw_data.txt)
